### PR TITLE
 Updated sphinx directives in "ipython_directive.rst"

### PR DESCRIPTION
**Pull request by LucianaMarques**
This pull request was made to make the changes mentioned in issue [#11410](https://github.com/ipython/ipython/issues/11410)

I did some improvements and was hoping that someone could review this.

I'm also having some trouble with git... I'm new to open source and forking repos, so I'm not sure how to correctly keep my remote repo updated with the original one. I did some things that I found online but not sure if everything is ok with this version I have.

Another thing, I wasn't familiar with rsStuctureText until today and I wanted to make some references within the document to a section (the decorators section), like a link or something. Is this possible?

Thank you!

### CODE REVIEW

**Comment by Carreau** [comment]
That looks good, we can merge that and iterate once the test are passing.

File: docs/source/development/ipython_directive.rst [hunk 1]
```diff
@@ -106,7 +116,25 @@ Multi-line input is supported.
...
    ['http://ichart.finance.yahoo.com/table.csv?s=CROX', 'd=9', 'e=22',
 
-You can do doctesting on multi-line output as well.  Just be careful
+Testing directive outputs
+=========================
+
+The IPython Sphinx Directive makes it possible to test the outputs that you provide with your code. To do this, 
+decorate the contents in your directive block with one of the following:
+
+  * list directives here
```
[the remark this answers is not in the document]
Comment (LucianaMarques) [reply docs/source/development/ipython_directive.rst]: I didn't know what to put here exactly. Do you have any suggestions?

File: docs/source/development/ipython_directive.rst [hunk 2]
```diff
@@ -134,6 +162,8 @@ suppress the seed line so it doesn't show up in the rendered output
           [0.22591016, 0.77731835],
           [0.0072729 , 0.34273127]])
 
+For more information on @supress and @dostest decorators, please refer to the end of this file in
```
[the remark this answers is not in the document]
Comment (LucianaMarques) [reply docs/source/development/ipython_directive.rst]: Nice! Should I make another commit?

**Comment by LucianaMarques** [comment]
Apparently one test didn't work for python 3.7

I looked at the test log, and it said this:

> 
> Exception occurred:
>   File "/home/travis/build/ipython/ipython/IPython/sphinxext/ipython_directive.py", line 870, in custom_doctest
>     doctest_type = args[1]
> IndexError: list index out of range

Is this anything known previously? If not, can I open an issue and try to solve it?

Thanks

**Comment by Carreau** [comment]
> Is this anything known previously? If not, can I open an issue and try to solve it?

Yes, you can try to solve it. I recently made the directive turn warnings into errors.
Let me know if you have difficulty narrowing down the bug, I can give a hand.

File: docs/source/development/ipython_directive.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (Carreau) [reply docs/source/development/ipython_directive.rst]: I'll have to reread the code, but my guess it's `@doctest` or `@savefig`.

File: docs/source/development/ipython_directive.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (Carreau) [reply docs/source/development/ipython_directive.rst]: Sure, or you can apparently "accept" in the GitHub UI. But it's brand new so I haven't used it before.

File: docs/source/development/ipython_directive.rst [hunk 3]
```diff
@@ -68,21 +71,24 @@ left off.
...
+match on a doctest assertion, an error will occur.
 
 
 .. ipython::
 
    In [1]: x = 'hello world'
 
    # this will raise an error if the ipython output is different
-   @doctest
+   @doctest"
```
Comment (Carreau) [inline docs/source/development/ipython_directive.rst]: I think the error is just this trailing quote.

Comment (Carreau) [reply docs/source/development/ipython_directive.rst]:
```suggestion
   @doctest
```

**Comment by LucianaMarques** [comment]
I accepted your observations!

**Comment by LucianaMarques** [comment]
There's more than one commit, though. Which is the best way to update my repo with these commits?
I'm getting lost with the commands haha

**Comment by LucianaMarques** [comment]
Hmm and the tests are not running. Should I try to push from my remote?

**Comment by Carreau** [comment]
There is/was a github outage, hence why things are not working properly. Some of the changes on the repo have been undone, and messages have disappeared. The best is too keep an eye on https://status.github.com/messages until all is green and then start reworking on things.

**Comment by LucianaMarques** [comment]
Hey @Carreau ! Thank you for merging my PR. 

I didn't understand, though, what happened. The problem was happening due to merge conflicts? Was this nothing to do with python version or anything else?

**Comment by Carreau** [comment]
> The problem was happening due to merge conflicts? Was this nothing to do with python version or anything else?

Yes and no. There was still som conflict where merged marked like the following were still there:

```
<<<<<<<<< one head
@doctest""
=========
@doctest 
>>>>>>>>>>>>> other head
```

and we build the documentation only on Python 3.7, so it was failing only on Python 3.7 :-)

**Comment by LucianaMarques** [comment]
Oh ok, I'll try to be more organized for the next pull requests. Thank you so much for being so patient with me! :-)

**Comment by Carreau** [comment]
> Thank you so much for being so patient with me! :-)

Always, you've done a great job and are already way better than me when I started ! (I was adding semicolon at the end of lines... ) And you've dome of lot of work and freeing us time to do more complex task. I hope in a couple of weeks you'll fix my mistakes !

**Comment by willingc** [comment]
Great work @LucianaMarques. We appreciate the contribution. :tada: :cake: :snake:

**Pull request merged**